### ISSUE TITLE
Generalize AWS credential settings

### ISSUE THREAD

**Issue by makhdumi** [sematext/logagent-js#98]
The elasticsearch-query input is perfect for what I need. I have to make a small modification to it though to connect to our AWS Elasticsearch cluster.

Since you're also planning to add an S3 input plugin, would it make sense to add general AWS connection (auth) settings, that could then be used by stuff like the input/elasticsearch-query and output/elasticsearch (with I guess output/aws-elasticsearch being removed)?

**Comment by otisg** [comment]
I think that would work.... If you don't mind, a PR would be great.

**Comment by megastef** [comment]
Yes. Plugins could access global settings via config.configFile.aws.someProperty

But plugins should also have the option to have the settings in the plugin section - this allows multiple instances of the plugin with different sources/destinations. So IF there is no local config the plugins could check the global section for the missing info.

**Comment by megastef** [comment]
@makhdumi The follwing global AWS config is supported for bot elasticsearch-query and Amazon Elasticsearch output plugin, when module specific AWS config is missing. For the ELasticsearch Query plugin, you have to set awsEnabled:true to enable AWS settings. 
We just released v2.0.104. 

```
# global credentials for Amazon Elasticsearch
aws:
  auth: 'username:password'
  awsConfigFile: ./aws-config.json

input:
  queryLogs: 
  module: elasticsearch-query
    # set to true for Amazon Elasticsearch service
    awsEnabled: true
    #auth: 'username:password'
    #awsConfigFile: ./aws-config.json
   ...
```

**Comment by makhdumi** [comment]
Thank you. This is awesome.
